builtin/logical/consul/path_token.go

```diff
@@ -84,39 +98,71 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 			"token": token,
 			"role":  role,
 		})
-		s.Secret.TTL = result.TTL
-		s.Secret.MaxTTL = result.MaxTTL
+		s.Secret.TTL = roleConfigData.TTL
+		s.Secret.MaxTTL = roleConfigData.MaxTTL
 		return s, nil
 	}
 
 	// Create an ACLToken for Consul 1.4 and above
-	policyLink := []*api.ACLTokenPolicyLink{}
-	for _, policyName := range result.Policies {
-		policyLink = append(policyLink, &api.ACLTokenPolicyLink{
-			Name: policyName,
-		})
+	// If policies were supplied here, then overwrite the policies
+	// that were given when the role was written
+	policies := d.Get("policies").([]string)
+	var policyLinks []*api.ACLTokenPolicyLink
+	if len(policies) > 0 {
+		policyLinks = getPolicies(policies)
+	} else {
+		policyLinks = getPolicies(roleConfigData.Policies)
+	}
+
+	// If a namespace was supplied here, then overwrite the namespace
+	// that was given when the role was written
+	namespace := d.Get("consul_namespace").(string)
+	if namespace == "" {
+		namespace = roleConfigData.Namespace
+	}
+	// If a partition was supplied here, then overwrite the partition
+	// that was given when the role was written
+	partition := d.Get("partition").(string)
+	if partition == "" {
+		partition = roleConfigData.Partition
```

Comment:
I don't think that we should allow these parameters to be specified during creds generation. Though exposing `policies`, `consul_namespace`, and `partition` on credentials (Consul token) generation can provide flexibility by allowing roles to be re-used, this same flexibility makes it really difficult for Vault to guard how a token should be generated based on values specified in the role.

The pattern for most, if not all, secrets engines is that the credentials generation endpoint takes nothing but the role name. This allows operators to be explicit on the attributes that are tied to the creds to always be consistent and always originate from role parameters. It also makes it easier to protect via Vault's ACL those attributes are tied to the role endpoint which is separate from the credentials generation endpoint.

Reply:
Removed the ability to do this in https://github.com/hashicorp/vault/pull/13850/commits/443e3e68fa3e880e3af23bbfee71692734df1395